### PR TITLE
Go dump route

### PR DESCRIPTION
**Pull request by iulianpascalau**
added `node/godump` api route able to send through logger go routines stack trace.

### CODE REVIEW

File: api/node/routes.go [hunk 1]
```diff
@@ -51,9 +55,21 @@ func Routes(router *gin.RouterGroup) {
 	router.GET("/statistics", Statistics)
 	router.GET("/status", StatusMetrics)
 	router.GET("/p2pstatus", P2pStatusMetrics)
+	router.GET("/godump", GoRoutinesDump)
```
Comment (andreibancioiu) [inline api/node/routes.go]: Perhaps rename to `/goroutines-dump`?

File: api/node/routes.go [hunk 2]
```diff
@@ -51,9 +55,21 @@ func Routes(router *gin.RouterGroup) {
...
 	// placeholder for custom routes
 }
 
+// GoRoutinesDump will dump go routines info
+func GoRoutinesDump(c *gin.Context) {
+	log.Warn("go routines dump")
+
+	buff := make([]byte, 1<<27)
```
Comment (andreibancioiu) [inline api/node/routes.go]: :)

`buffSize := 128 * 1024 * 1024`, for improved readability? Make it **smaller**? 128 MB is a bit much to dump in the logger anyway.

File: api/node/routes.go [hunk 3]
```diff
@@ -51,9 +55,21 @@ func Routes(router *gin.RouterGroup) {
...
+	num := runtime.Stack(buff, true)
+	log.Warn(string(buff[:num]))
+
+	c.JSON(http.StatusOK, gin.H{"go routines dump": true})
```
Comment (andreibancioiu) [inline api/node/routes.go]: Why not respond with the dump?

Comment (iulianpascalau) [reply api/node/routes.go]: will be overkill for the http server

File: api/node/routes.go (hunk 2 above)
Comment (iulianpascalau) [reply api/node/routes.go]: We had a previous situation where we had thousands of go routines. Having a large buffer will allow us to tell which go routines repeat frequently (that can be a sign of a go routine leakage). This is just the maximum buffer, when logging I do a slice. Will change to 128 * 1024 * 1024 for readability concerns.

File: api/node/routes.go (hunk 1 above)
Comment (iulianpascalau) [reply api/node/routes.go]: to much to type and memorize. I would like to keep it as simple as possible

**Comment by iulianpascalau** [comment]
needs more refactoring

**Pull request closed without merging**